tools/moFF/moff.xml

```diff
@@ -0,0 +1,400 @@
+                    echo -e "peptide\tprot\tmod_peptide\trt\tmz\tmass\tcharge" > './tempfile${i}_2.tab' &&
+                    awk 'BEGIN{OFS="\t"; FS="\t"}{print \$pep,\$prot,\$mod,\$rt,\$mz,\$mass,\$charge}' pep="${task.ident_input.peptide}" prot="$task.ident_input.prot" mod="$task.ident_input.mod_peptide" rt="$task.ident_input.rt" mz="$task.ident_input.mz" mass="$task.ident_input.mass" charge="$task.ident_input.charge" './tempfile${i}_1.tab' >> './tempfile${i}_2.tab' &&
+                    mv './tempfile${i}_2.tab' './ident_inputs/${value.element_identifier}$format' &&
+                #end for
+            #end if
+        ]]></token>
+        <token name="@RAW_INPUT_ARG_SINGLE@"><![CDATA[
+            --raw_list './raws/$task.msms_input.raw_list.element_identifier$format'
+        ]]></token>
+        <token name="@RAW_INPUT_ARG_MULTIPLE@"><![CDATA[
+            --raw_list
+            #for $value in $task.msms_input.raw_list:
+                './raws/$value.element_identifier$format'
+            #end for
+        ]]></token>
+        <token name="@WRANGLE_RAW_INPUT_SINGLE@"><![CDATA[
+            mkdir ./raws &&
+            ## for files, need to softlink the name to the history item
+            ln -s '$task.msms_input.raw_list' './raws/$task.msms_input.raw_list.element_identifier$format' &&
+        ]]></token>
+        <token name="@WRANGLE_RAW_INPUT_MULTIPLE@"><![CDATA[
+            mkdir ./raws &&
+            ## for files, need to softlink the name to the history item
+            #for $value in $task.msms_input.raw_list:
+                ln -s '$value' './raws/$value.element_identifier$format' &&
+            #end for
+        ]]></token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@VERSION@">moff</requirement>
+    </requirements>
+    <version_command>echo @VERSION@</version_command>
+    <command detect_errors="aggressive"><![CDATA[
+        @FORMAT@
+        mkdir ./out &&
+        #if $task.task_selector == "moff":
+            @WRANGLE_IDENT_INPUT_SINGLE@
+            @WRANGLE_RAW_INPUT_SINGLE@
+            moff_all.py
+                @IDENT_INPUT_ARG_SINGLE@
+                @RAW_INPUT_ARG_SINGLE@
+                --tol $task.tol
+                --mbr $task.mbr
+                --xic_length $task.xic_length
+                --rt_peak_win $task.rt_peak_win
+	        --loc_out ./out
+                    #if ($task.match_filter.match_filter_selection):
+                        --match_filter
+                        --sample_size $task.match_filter.sample_size
+                        --quantile_thr_filtering $task.match_filter.quantile_thr_filtering
+                        #if ($task.match_filter.ptm_file):
+                            --ptm_file '$task.match_filter.ptm_file'
+                        #else:
+                            --ptm_file ./ptm_setting_ps.json
```

Comment:
here , we should specify the path for the default ptm file. Actually the file it is in the moFF repository, I dont  know which the path of the file when it is executed inside moFF conda enviroment.  Probably the path used here is broken for sure.